### PR TITLE
Exception handling / it-academy.by / JC1-110-22 / Maksim Bondar / Tasks 1 - 7 complete

### PR DESCRIPTION
**Pull request by shiv-qa-auto**
## Pull request state

Actual status for PR:
- [x] implemented
- [x] reviewed
- [x] fixed
- [ ] approved

### CODE REVIEW

File: src/main/java/com/rakovets/course/java/core/practice/exception_handling/Task4.java [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package com.rakovets.course.java.core.practice.exception_handling;
+
+public class Task4 {
+
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/exception_handling/Task4.java:4]: "Violation: Useless blank lines
Solution: Remove blank lines"

**Review by rakovets — CHANGES_REQUESTED** [review]
Need full refactoring:

Task - throw exception
Demo - catch exception

File: src/main/java/com/rakovets/course/java/core/practice/exception_handling/Task1.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package com.rakovets.course.java.core.practice.exception_handling;
+
+public class Task1 {
+
```
[the remark this answers is not in the document]
Comment (shiv-qa-auto) [reply src/main/java/com/rakovets/course/java/core/practice/exception_handling/Task1.java]: Fixed, please review

File: src/main/java/com/rakovets/course/java/core/practice/exception_handling/Task2.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package com.rakovets.course.java.core.practice.exception_handling;
+
+public class Task2 {
+
```
[the remark this answers is not in the document]
Comment (shiv-qa-auto) [reply src/main/java/com/rakovets/course/java/core/practice/exception_handling/Task2.java]: Fixed, please review

File: src/main/java/com/rakovets/course/java/core/practice/exception_handling/Task3.java [hunk 4]
```diff
@@ -0,0 +1,11 @@
+package com.rakovets.course.java.core.practice.exception_handling;
+
+public class Task3 {
+
```
[the remark this answers is not in the document]
Comment (shiv-qa-auto) [reply src/main/java/com/rakovets/course/java/core/practice/exception_handling/Task3.java]: Fixed, please review

File: src/main/java/com/rakovets/course/java/core/practice/exception_handling/Task5.java [hunk 5]
```diff
@@ -0,0 +1,15 @@
+package com.rakovets.course.java.core.practice.exception_handling;
+
+public class Task5 extends Task4 {
+
```
[the remark this answers is not in the document]
Comment (shiv-qa-auto) [reply src/main/java/com/rakovets/course/java/core/practice/exception_handling/Task5.java]: Fixed, please review

**Review by rakovets — APPROVED** [review]

**Pull request closed without merging**